src/drivers/gps/gps.cpp

```diff
@@ -1068,6 +1072,43 @@ gps_main(int argc, char *argv[])
 			}
 		}
 
+		/* Detect interface option */
+		for (int i = 2; i < argc; i++) {
+			if (!strcmp(argv[i], "-i")) {
+
+				int interface_arg = i + 1;
+
+				if (interface_arg < argc) {
+					if (!strcmp(argv[interface_arg], "spi")) {
+						interface = GPS_DRIVER_SPI;
+
+					} else if (!strcmp(argv[interface_arg], "uart")) {
+						interface = GPS_DRIVER_UART;
+					}
+				}
+			}
+		}
+
+		/* Detect mode/protocol option */
+		for (int i = 2; i < argc; i++) {
+			if (!strcmp(argv[i], "-p")) {
+
+				int mode_arg = i + 1;
+
+				if (mode_arg < argc) {
+					if (!strcmp(argv[mode_arg], "ubx")) {
```

Comment:
The change here should have UBX, MTK, ASHTECH and AUTO, with auto being the default.